tools/shared/src/main/scala/org/scalajs/core/tools/linker/checker/IRChecker.scala

```diff
@@ -293,9 +312,11 @@ private final class IRChecker(unit: LinkingUnit, logger: Logger) {
             "but must be Any")
       }
 
-      val thisType =
-        if (classDef.kind.isJSClass) AnyType
+      val thisType = {
+        if (isTopLevel) NoType
```

Comment:
Consider `if (static)`, which is the actually reason `thisType` is `NoType`. The fact that `static == isTopLevel` is an accident, from the point of view of `thisType`.